core_backend/app/db/db_models.py

```diff
@@ -316,71 +321,258 @@ class ContentDB(Base):
+
+async def get_list_of_languages_from_db(
+    asession: AsyncSession, offset: int = 0, limit: Optional[int] = None
+) -> List[LanguageDB]:
+    """
+    Retrieves all content from the database
+    """
+    stmt = select(LanguageDB)
+    if offset > 0:
+        stmt = stmt.offset(offset)
+    if limit is not None:
+        stmt = stmt.limit(limit)
+
+    language_rows = (await asession.execute(stmt)).all()
+
+    return [c[0] for c in language_rows] if language_rows else []
+
+
+async def is_language_name_unique(language_name: str, asession: AsyncSession) -> bool:
+    """
+    Check if the language name is unique
+    """
+    stmt = select(LanguageDB).where(LanguageDB.language_name == language_name)
+    language_row = (await asession.execute(stmt)).scalar_one_or_none()
+    if language_row:
+        return False
+    else:
+        return True
+
+
+class ContentTextDB(Base):
+    """
+    SQL Alchemy data model for content text
+    """
+
+    __tablename__ = "content_texts"
+
+    content_text_id: Mapped[int] = mapped_column(
+        Integer, primary_key=True, nullable=False
+    )
     content_embedding: Mapped[Vector] = mapped_column(
         Vector(int(PGVECTOR_VECTOR_SIZE)), nullable=False
     )
+
     content_title: Mapped[str] = mapped_column(String(length=150), nullable=False)
     content_text: Mapped[str] = mapped_column(String(length=2000), nullable=False)
-    content_language: Mapped[str] = mapped_column(String, nullable=False)
-
-    content_metadata: Mapped[JSONDict] = mapped_column(JSON, nullable=False)
-
     created_datetime_utc: Mapped[datetime] = mapped_column(DateTime, nullable=False)
     updated_datetime_utc: Mapped[datetime] = mapped_column(DateTime, nullable=False)
+    content_metadata: Mapped[JSONDict] = mapped_column(JSON, nullable=False)
+
+    language_id: Mapped[int] = mapped_column(
+        Integer, ForeignKey("languages.language_id"), nullable=False
+    )
+
+    content_id: Mapped[int] = mapped_column(
+        Integer, ForeignKey("contents.content_id"), nullable=False
+    )
+
+    language: Mapped["LanguageDB"] = relationship("LanguageDB")
+
+    content: Mapped["ContentDB"] = relationship("ContentDB")
 
     def __repr__(self) -> str:
         """Pretty Print"""
-        return f"<Content #{self.content_id}:  {self.content_text}>"
+        return f"<ContentText #{self.content_text_id}:  {self.content_text}>"
+
+
+async def is_content_language_combination_unique(
+    content_id: int, language_id: int, asession: AsyncSession
+) -> bool:
+    """
+    Check if the content and language combination is unique
+    """
+    stmt = select(ContentTextDB).where(
+        (ContentTextDB.content_id == content_id)
+        & (ContentTextDB.language_id == language_id)
+    )
+    content_row = (await asession.execute(stmt)).scalar_one_or_none()
+    if content_row:
+        return False
+    else:
+        return True
 
 
 async def save_content_to_db(
-    content: ContentCreate,
+    content: ContentTextCreate,
```

Comment:
Should we be explicit and name the function and variable `content_text`?

Reply:
Should we also allow creating/deleting an entire content rather than language-by-language? cc: @Tanmay-97

Reply:
On the last point, we discussed that for now we'll keep it this way. Limiting interactions with content table seems to make sense. Can make any necessary changes later